src/main/js/common/api.js

```diff
@@ -4,42 +4,35 @@
 
 import { getJSON, postJSON, post } from "sonar-request";
 
-// Function used to get current SonarQube Server version
+//Function used to get current SonarQube Server version
 function getSonarVersion() {
   return getJSON("/api/system/status").then(response => {
     return parseFloat(response.version);
   });
 }
 
-// Function used to get the list of existing projects (until 7.9.x)
-function getProjectsList79() {
-  return getJSON("/api/projects/index").then(response => {
-    return response.map(item => {
-      // Attribute names changed in SonarQube 8.0
-      // so we need to match the new ones for the frontend
-      item.key = item.k;
-      item.name = item.nm;
-      return item;
-    });
-  });
-}
-
-// Function used to get the list of existing projects (since 8.0)
-function getProjectsList8() {
-  return getJSON("/api/projects/search").then(response => {
-    return response.components;
-  });
-}
-
-// Function used to get the list of existing project regarding SonarQube version
-export function getProjectsList() {
-  return getSonarVersion().then(version => {
-    // Depending on SonarQube version, the API is not the same to retrieve projects list
-    if (version >= 8.0) {
-      return getProjectsList8();
-    } else {
-      return getProjectsList79();
+//Functions used to get all user projects
+export function getProjectsList(){
+  let elementByPage = 500;
+  let allPromises = [];
+  let finalProjectsList = [];

```

Comment:
See my comment below regarding that same variable, bu it may not be useful anymore.